typings/index.d.ts

```diff
@@ -3459,6 +3496,31 @@ declare module 'discord.js' {
 
   type MessageResolvable = Message | Snowflake;
 
+  interface MessageSelectMenuOptions extends BaseMessageComponentOptions {
+    customID?: string;
+    disabled?: boolean;
+    maxValues?: number;
+    minValues?: number;
+    options?: MessageSelectOptionData[];
+    placeholder?: string;
+  }
```

Comment:
why are these all marked as optional? looks like BaseMessageComponentOptions and MessageButtonOptions have the same issue
also, should probably enforce the `type` on the specific component options

```suggestion
  interface MessageSelectMenuOptions extends BaseMessageComponentOptions {
    customID: string;
    disabled?: boolean;
    maxValues?: number;
    minValues?: number;
    options: MessageSelectOptionData[];
    placeholder?: string;
  }
```

Reply:
Because it's not required to pass any of these to the constructors for a MessageButton or MessageSelectMenu

Reply:
These options interfaces can also be used directly when sending a message, in which case the fields are required; the constructors should use Partial<...Options>

Reply:
Fair point, I've been considering similar things - arguably if passing options to a constructor it *should* enforce all the required params, though in that instance `type` can be omitted, while for raw formats in messages it can't be. I wont commit this change just yet, will review the types more broadly later.

Reply:
Yeah, I'm going to leave this out of this PR, since if I'm going to make typings changes like this it should be consistent with Buttons too. Will do separately for both after this merges.